openembedded-core-mbl/scripts/signctl/README.md

```diff
@@ -0,0 +1,329 @@
+
+2. "Activate" the environment.
+```
+$ source /path/to/venv/bin/activate
+```
+
+3. If you used the Ubuntu system python as the base as shown in the example command you must upgrade the pip installation in the venv. This is because the version of pip that comes with Ubuntu 16.04 doesn't have support for Python wheels.
+
+```
+$ python -m pip install --upgrade pip
+```
+
+4. Navigate to the `meta-mbl/openembedded-core-mbl/scripts/signing/signctl` folder.
+
+Install from source using `pip`.
+```
+$ pip install .
+```
+
+## Signing Tool Command Line Interface
+```
+usage: signctl.py [-h] [--backend-url BACKEND_URL] {generate,sign-atf} ...
+
+positional arguments:
+  {generate,sign-atf}
+    generate            Generate a keychain and store it in the storage
+                        backend.
+    sign-atf            Sign ATF BL and FIP.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --backend-url BACKEND_URL
+```
+```
+usage: signctl.py generate [-h] [--cot-keys] [--hab-srks]
+                           [--output-dir OUTPUT_DIR]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --cot-keys            Generate an ATF chain of trust keychain.
+  --hab-srks            Generate super root public keys for NXP HAB.
+  --output-dir OUTPUT_DIR
+                        The output directory for the HAB SRKs (public part).
+```
+```
+usage: signctl.py sign-atf [-h] [--rpi-armstub8 RPI_ARMSTUB8] --fip FIP
+                           [FIP ...] [--patch-rotkey]
+                           [--output-dir OUTPUT_DIR]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --rpi-armstub8 RPI_ARMSTUB8
+                        Path to armstub8.bin
+  --fip FIP [FIP ...]   Path to one or more FIPs to be signed.
+  --patch-rotkey        Patch root of trust public key hash in to BL1 and BL2.
+  --output-dir OUTPUT_DIR
+                        Output directory for signed images and the original
+                        fip components.
+```
+
+## Generating Chain of Trust Keychain
+
+You can generate the keychain used for the ATF CoT using the `signctl generate --cot-keys` command.
+
+Currently the `generate --cot-keys` command creates all of the required CAs up to the root of trust. The private keys are stored in the Vault.
+
+In the release version, the `signctl` tool will support generating individual keypairs in the keychain, allowing for key revocation and access restriction to high security keys.
+
+<span class="notes">**Note:** The tool only supports signing for a single device at the moment. The release version will support multiple devices; we will provide documentation and guidance for setting access control policies in Vault, and for locking down the Vault machine for production use. **The current flow is only for development use.**</span>
```

Comment:
Could add this as a point in the overview section on restrictions above.

Reply:
Moved to the overview section in bd16b66